### DISCUSSION TITLE
Need Help with Selection Scope settings

### DISCUSSION THREAD

**Discussion by oussamabonnor1** [iTwin/itwinjs-core#4718] [Q&A]
We're trying to default our selection scope to `Element` by using 

```javascript
UiFramework.setActiveSelectionScope("element")
```

in our custom Frontstage component.

By using
```javascript
 iFramework.getActiveSelectionScope()
```
We get `Element` as result but the selection is always top-assembly based.

After a selection, we logged the scope again and found that it's reverting back to `top-assembly`.

We also tried to update the `Presentation.selection.scopes.activeScope` but the same result is taking place.

Our best guess is that [this](https://github.com/iTwin/itwinjs-core/blob/9d29d30c5d2a494dc92b3c46a445fc3ea60f2620/ui/appui-react/src/appui-react/UiFramework.ts#L354) is not working properly.

Any other way we could programmatically default the selection scope to element ?

**Comment by NancyMcCallB** [comment]
You are correct that it is either not correctly persisting to state or not initializing from state properly. I'm looking at it now.
Thanks!

**Reply by NancyMcCallB** [reply]
It seems that the setting is being properly saved to the session state, but in a new session, the active scope is being loaded from Presentation based on the iModelConnection. I'm going to need a bit more time to figure out what's best to fix it.

**Reply by oussamabonnor1** [reply]
Thanks for looking into it, just wanted to point out [this sample](https://www.itwinjs.org/sandboxes/iTwinPlatform/Selection%20Scope%20-%20Geometric%20Elements) in case it could be of some help.

If you look in the `ElementPickerWidget.tsx` on line 75, there's an interesting comment about the Presentation Manager and the scopes it controls. Frankly I didn't understand it but it might help.

It goes as follows
```
The Presentation manager will always be set to element scope. 
When the active scope changes, instead of changing the real scope in the Presentation Manager, 
we will instead compute the scope logic ourselves.
```

**Reply by Alfonso-Martello** [reply]
Hello,

I wanted to add a reply clarifying that sample and comment. The sample itself is a little confusing as it does not use the Presentation Manager beyond that point. Instead, it aims to show how to manually calculate what would have been selected and highlighted had an element id been selected with a specific scope. The idea being that you can take a specific element id and calculate what would have been selected and highlighted using a certain scope without clicking on that element and without changing the scope of the Presentation Manager. For example, if someone wanted to have the Presentation Manager always be set to Element scope, but to show one specific element in Assembly scope without the user having to click on that element, they could do so by following the code in the sample.

A better comment at that point in the sample would have been:
```
We set the Presentation manager to element scope during initialization.
After this point, we never change the scope set in the Presentation Manager.
Instead, we track which scope is selected in the dropdown and calculate the
selection set and the hilite set manually based on that scope.
Technically, this sample never uses the selection scope set in the Presentation Manager.
Instead this line of code is kept for two reasons:
	1. To show how the Presentation Manager scope could be set. The selection scope of the Presentation Manager
	   could also be set by using the UiFramework getActiveSelectionScope and/or setActiveSelectionScope functions.
	2. To distinguish that this sample does not make use of the Presentation Manager beyond this point.
```

In terms of your original question, I defer to Nancy's expertise and investigation on the issue.

**Reply by oussamabonnor1** [reply]
Thank you @Alfonso-Martello, it does make sense now. I appreciate the input.

**Reply by oussamabonnor1** [reply]
On a separate note, @NancyMcCallB I'm not sure what you're doing but the selection scope is defaulted to element in our production app without us including a change! Let me know if the issue is now solved :)
